tests/TestCase/Controller/Component/RequestHandlerComponentTest.php

```diff
@@ -531,6 +520,26 @@ public function testNoViewClassExtension()
         $this->assertEquals('csv', $this->Controller->viewBuilder()->getLayoutPath());
     }
 
+    /**
+     * Tests that configured extensions that have no configured mimetype do not silently fallback to HTML.
+     *
+     * @return void
+     * @expectedException \Cake\Http\Exception\NotFoundException
+     * @expectedExceptionMessage Invoked extension not recognized/configured: foo
+     */
+    public function testUnrecognizedExtensionFailure()
+    {
+        Router::extensions(['json', 'foo'], false);
+        $this->Controller->request = $this->Controller->request->withParam('_ext', 'foo');
```

Comment:
Using setRequest()/getRequest() instead of the $request property will save use effort when doing 3.next/4.x merges :)

Reply:
You want me to touch the code again? :)